Tasks/ANTPreview/ant.ps1

```diff
@@ -50,17 +52,48 @@ if($jdkVersion -and $jdkVersion -ne "default")
 
     Write-Host "Setting JAVA_HOME to $jdkPath"
     $env:JAVA_HOME = $jdkPath
-    Write-Verbose "JAVA_HOME set to $env:JAVA_HOME"
+    Write-Verbose "JAVA_HOME set to $env:JAVA_HOME" -Verbose
 }
 
-Write-Verbose "Running Ant..."
+$buildRootPath = Split-Path $antBuildFile -Parent
+$reportDirectoryName = "CodeCoverage"
+$reportDirectory = Join-Path $buildRootPath $reportDirectoryName
+
+if(Test-Path $reportDirectory)
+{
+   # delete any previous code coverage data 
+   rm -r $reportDirectory -force 
+}
+
+$summaryFileName = "summary.xml"
+$summaryFile = Join-Path $buildRootPath $reportDirectoryName 
+$summaryFile = Join-Path $summaryFile $summaryFileName
+# ensuring unique code coverage report task name by using guid
+$CCReportTask = "CodeCoverage_" +[guid]::NewGuid()
+
+if($isCoverageEnabled)
+{
+   # Enable code coverage in build file
+   Enable-CodeCoverage -BuildTool 'Ant' -BuildFile $antBuildFile -CodeCoverageTool $codeCoverageTool -ClassFilter $classFilter -ClassFilesDirectory $classFilesDirectory -SourceDirectory $srcDirectory -SummaryFile $summaryFileName -ReportDirectory $reportDirectoryName -CCReportTask $CCReportTask
+   Write-Verbose "code coverage is successfully enabled." -Verbose
+}
+	
+
+Write-Verbose "Running Ant..." -Verbose
 Invoke-Ant -AntBuildFile $antBuildFile -Options $options -Targets $targets
 
+if($isCoverageEnabled)
+{
+   # run report code coverage task which generates code coverage reports.
+   Write-Verbose "Reporting code coverage" -Verbose
+   Invoke-Ant -AntBuildFile $antBuildFile -Targets $CCReportTask
+}
+
 # Publish test results files
 $publishJUnitResultsFromAntBuild = Convert-String $publishJUnitResults Boolean
 if($publishJUnitResultsFromAntBuild)
 {
-   # check for JUnit test result files
+    # check for JUnit test result files
     $matchingTestResultsFiles = Find-Files -SearchPattern $testResultsFiles
     if (!$matchingTestResultsFiles)
     {
```

Comment:
I think this doesnt suite  warning bucket. As ant task can be used just for build also. thoughts?